src/clj_http/conn_mgr.clj

```diff
@@ -232,20 +247,26 @@
       get-custom-strategy-registry))
 
 (defn ^BasicHttpClientConnectionManager make-regular-conn-manager
-  [{:keys [keystore trust-store
+  [{:keys [custom-dns-resolver
+           keystore trust-store
            key-managers trust-managers
            socket-timeout] :as req}]
   (let [conn-manager (cond
                        (or key-managers trust-managers)
-                       (BasicHttpClientConnectionManager. (get-managers-scheme-registry req))
+                       (BasicHttpClientConnectionManager. (get-managers-scheme-registry req)
+                                                          nil nil
+                                                          (when custom-dns-resolver (dns-custom-resolver custom-dns-resolver)))
 
                        (or keystore trust-store)
-                       (BasicHttpClientConnectionManager. (get-keystore-scheme-registry req))
+                       (BasicHttpClientConnectionManager. (get-keystore-scheme-registry req)
+                                                          nil nil
+                                                          (when custom-dns-resolver (dns-custom-resolver custom-dns-resolver)))
```

Comment:
same suggestion as above, remove the wrapper and pass in the dns-resolver directly